Publish Sep 2025 newsletter (#476)
maplibre/maplibre.github.io: src/content/news/2025-10-04-maplibre-newsletter-september-2025/index.mdx

Other improvements this month include:

- Fixes for raster tile fading, with more enhancements coming soon thanks to Ryan ([@wayofthefuture](https://github.com/wayofthefuture)).
- Numerous fixes to [`global-state`](https://maplibre.org/maplibre-style-spec/expressions/#global-state), primarily led by [@melitele](https://github.com/melitele), significantly ironing out most of the related issues.

## 📝 MapLibre Tile Specification

The [maplibre/maplibre-tile-spec](https://github.com/maplibre/maplibre-tile-spec) project is nearing its first release, focused on functional compatibility with [MVT](https://wiki.openstreetmap.org/wiki/Vector_tiles).

[MLT (MapLibre Tile)](https://maplibre.org/roadmap/maplibre-tile-format/) is a new vector tile format designed for efficient storage and rendering of map data, optimized for the MapLibre ecosystem.

> Current implementations include:

Reviewer: ```suggestion
We have added an [implementation status overview page](https://maplibre.org/maplibre-tile-spec/implementation-status/) to the documentation site of the MapLibre Tile Specification. Here you can track the status of (planned) implementations and integrations, or add your own.
```

Reviewer: Better to link to a page that will stay up to date. And that has links.

> - Java: encoder (research quality)

Reviewer: I would remove these.

- C++: decoder (in progress, targeting production quality)
- JavaScript: decoder (research quality)
- Rust: encoder and decoder (research quality)

## ✏️ Maputnik

This month has been a very productive one for [our small style editor](https://maputnik.github.io/).

Maputnik had recently gained the status of a ["supported project"](https://github.com/maplibre/maplibre/blob/main/PROJECT_TIERS.md#supported-project) in MapLibre, so if there's something you'd like to work on and get paid this is now possible.

> We have been working hard to update all the libraries to their latest versions, including React to version 19. This upgrade required replacing outdated libraries that are no longer maintained and fixing issues across the existing codebase.

Reviewer: ```suggestion
Harel has been working hard to update all the libraries to their latest versions, including React to version 19. This upgrade required replacing outdated libraries that are no longer maintained and fixing issues across the existing codebase.
```

I think @HarelM should get the credit.

- Server-side style raster tile rendering
- PMTiles sources for more cloud providers, beyond just AWS S3

We’re on track to release Martin v1.0 by the FOSS4G conference in November. You can follow the ongoing work in [this GitHub discussion](https://github.com/maplibre/martin/discussions/1805).

Like other MapLibre projects, Martin has also hardened its CI and release infrastructure against supply chain attacks in this release. We are now publishing immutable, attested releases via CI instead of manually.
Further details are available [on Slack](https://slack.openstreetmap.us/).

## 🦀 maplibre-native-rs

The MapLibre Native Rust bindings have seen a steady stream of refactorings and breaking changes to enable server-side rendering both in Martin and other projects possible.

We currently offer a single-instance Pool (MapLibre Native internally uses multithreading) for rendering tiles, which already provides solid performance.

Work is ongoing to design a multi-instance Pool, and we’re actively seeking feedback. If you have complex server-side rendering use cases, we’d love to hear from you to ensure the upcoming APIs work well for your needs.

> 

Reviewer 2: There's alternatively also a small video from the WebGPU pr that can be used

> 

Reviewer 3: Or both.. you are rather high up in the blog post, so adding more images is nice 😉

> 

Author: I will add the image for now.

---
title: "MapLibre Newsletter September 2025"
date: "2025-10-04"
categories: ["newsletter"]
authors:
  [
    harel-mazor,
    bart-louwers,
    birk-skyum,
    joscha-eckert,
    frank-elsinga,
    yuri-astrakhan,
    ramya-ragupathy,
  ]
draft: false
---

import { Image } from "astro:assets";

Welcome to the MapLibre Newsletter, September 2025 edition!

This edition features advances in MapLibre Native, GL JS, Flutter, Martin, and more. Plus, we’ve given [our roadmap section](https://maplibre.org/roadmap/) a fresh redesign. Go check it out to see what’s in progress, what’s under consideration, and what’s been released.

> 
> A big thanks to [Birk Skyum](https://maplibre.org/about/birk-skyum/), [Frank Elsinga](https://maplibre.org/about/frank-elsinga/) and [Yuri Astrakhan](https://maplibre.org/about/yuri-astrakhan/) for their continued leadership, contributions, and support in driving these initiatives forward. 🙌

Reviewer 3: Lets not do that, unless @birkskyum wants to be mentioned ^^

```suggestion
```

If at all, this is birks' work

```suggestion
A big thanks to [Birk](https://maplibre.org/about/birk-skyum/) for driving this initiative forward. 🙌
```

## 🌐 MapLibre GL JS

September was a busy month with four new releases:[5.7.1](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.7.1), [5.7.2](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.7.2), [5.7.3](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.7.3) and [5.8.0](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.8.0).

The latest one, 5.8.0, stands out, as it introduces data-driven styling for [line-dasharray](https://maplibre.org/maplibre-style-spec/layers/#line-dasharray), a long-awaited feature and [the most upvoted issue](https://github.com/maplibre/maplibre-gl-js/issues/1235) in our tracker. Thanks Lucas (@lucaswoj) for this great contribution! We are super excited someone in the community decided to work on it.

Other improvements this month include:

- Fixes for raster tile fading, with more enhancements coming soon thanks to Ryan ([@wayofthefuture](https://github.com/wayofthefuture)).
- Numerous fixes to [`global-state`](https://maplibre.org/maplibre-style-spec/expressions/#global-state), primarily led by [@melitele](https://github.com/melitele), significantly ironing out most of the related issues.

> ## 📝 MapLibre Style Specification

Reviewer 3: typo, but a fun one 😉

```suggestion
## 📝 MapLibre Tile Specification
```

- Android: [v11.13.5](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.13.5)
- Web: [v5.8.0](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.8.0)

A major ongoing goal for the package is to deliver a seamless cross-platform experience. A map style can now be added as a JSON string, via a Flutter assets path, or through a URI on all platforms. Additionally, PMTiles support on the web has been added to achieve feature parity with Android and iOS.

More functionality has made its way into this release. You can find all user facing changes in the [changelog](https://pub.dev/packages/maplibre/changelog#030).

## 🧩 Martin

This month, [Martin](https://martin.maplibre.org/) has undergone significant refactoring.

For those not following Martin’s versioning closely: before reaching v1.0, we are doing SemVer-breaking releases for each minor version (every X in 0.X.Y), since Martin is also published as a Rust library, allowing you to build your own tile server using its internals.

We’ve now refactored the core functionality into a separate library, [`martin-core`](http://lib.rs/martin-core).

Other highlights this month include:

> - New warnings for misspelled config keys.
> - A brand new logo 🎉.

Reviewer 3: lets call out good non-maintainer work, I forgot about this 😉

```suggestion
- New warnings for misspelled config keys.
- A brand new logo thanks to Ryan ([@wayofthefuture](https://github.com/wayofthefuture)) 🎉.
```
